lib/rules/curly.js

```diff
@@ -309,6 +309,98 @@ module.exports = {
                 hasUnsafeIf(statement) && isFollowedByElseKeyword(node);
         }
 
+        /**
+         * Examples:
+         *
+         * The following location should be reported
+         *
+         * if (foo) bar()
+         *         ^
+         *
+         * while(foo) bar();
+         *           ^
+         *
+         * for(;;) bar();
+         *         ^
+         *
+         * for (var foo of bar) console.log(foo)
+         *                      ^
+         *
+         * for (var foo in bar) console.log(foo)
+         *                      ^
+         *
+         * Get the location of missing parent location where it should have been placed.
+         * @param {ASTNode} node the node to check.
+         * @param {string} name name to check.
+         * @returns {Object} Position object.
+         */
+        function getMissingLoc(node, name) {
```

Comment:
There's a lot of duplicated logic between these two methods. Could we combine them into one and pass a parameter indicating whether the curly braces are missing or not?

```js
function getReportLoc(node: ASTNode, name: string, isMissing: boolean) {}

// or, to avoid the boolean trap, we could also use an options object:

function getReportLoc(
    { node, name, isMissing }: { node: ASTNode, name: string, isMissing: boolean }
) {}
```

Reply:
> function getReportLoc(
    { node, name, isMissing }: { node: ASTNode, name: string, isMissing: boolean }
) {}

I didnt get this, so I went with the first one if its ok !.

let me know if this needs to be changed.

Reply:
No worries! It looks a bit confusing, but this is indicating type definitions for a destructured object.

Reply:
opz, I got it, I was thinking as a point of `js` .

should I change this?